src/azul/indexer/log_forwarding_service.py

```diff
@@ -0,0 +1,110 @@
+import csv
+import gzip
+from typing import (
+    Iterator,
+)
+
+from botocore.response import (
+    StreamingBody,
+)
+from furl import (
+    urlsplit,
+)
+
+from azul import (
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.types import (
+    MutableJSON,
+)
+
+
+class LogForwardingService:
+    """
+    Parse logs from an Application Load Balancer.
+    Adapted from https://github.com/rupertbg/aws-load-balancer-logs-to-cloudwatch
+    """
+
+    def read_logs(self, bucket: str, key: str) -> Iterator[MutableJSON]:
+        response = aws.s3.get_object(Bucket=bucket, Key=key)
+        body: StreamingBody = response['Body']
+        with gzip.open(body, mode='rt') as f:
```

Comment:
I think we've established by now that the logs are not encoded in UTF-8. So either

```suggestion
        with gzip.open(body, mode='rt', encoding='ascii') as f:
```

or 

```suggestion
        with gzip.open(body, mode='rt', encoding='latin1') as f:
```